pkg/cmd/pulumi/util.go

```diff
@@ -882,3 +882,30 @@ func buildStackName(stackName string) (string, error) {
 
 	return stackName, nil
 }
+
+// we only want to log a secrets decryption for a service backend project
+// we will allow any secrets provider to be used (service or self managed)
+// we will log the message and not worry about the response. The types
+// of messages we will log here will range from single secret decryption events
+// to requesting a list of secrets in an individual event e.g. stack export
+// the logging event will only happen during the `--show-secrets` path within the cli
+func log3rdPartySecretsProviderDecryptionEvent(ctx context.Context, backend backend.Stack,
+	secretName, commandName string) {
```

Comment:
Nit: Does it ever make sense to have both `secretName` and `commandName` set, or are these mutually exclusive?

It seems like it'd be somewhat easy mix-up and set the wrong parameter. It might be worth having separate functions for the callers to call. Or at a minimum, perhaps using some asserts, e.g.:

```go
if secretName != "" {
	contract.Assert(commandName == "");
	_ = client.Log3rdPartySecretsProviderDecryptionEvent(ctx, id, secretName)
}

if commandName != "" {
	contract.Assert(secretName == "");
	_ = client.LogBulk3rdPartySecretsProviderDecryptionEvent(ctx, id, commandName)
}
```

Reply:
I really like this! I will get tis updated now